src/Either.js

```diff
@@ -70,6 +70,23 @@ _Right.prototype.chain = function(f) {
   return f(this.value);
 };
 
+//chainRec
+Either.chainRec = Either.prototype.chainRec = function(f, i) {
+  var state = util.chainRecNext(i);
+  while (state.done === false) {
+    state = Either.either(
+      function(v) {
+        return { done: true, value: v, isLeft: true };
+      },
+      function(v) {
+        return v;
+      },
+      f(util.chainRecNext, util.chainRecDone, state.value)
```

Comment:
I have updated my implementation but I think it's better to use custom structure for representing state instead of Either and it's also simpler to understand.